week09/124. Binary Tree Maximum Path Sum.cpp

```diff
@@ -0,0 +1,35 @@
+//TC: O(N)
+//MC: O(H), H -> max height of the tree
+
+class Solution {
+public:
+    int maxSum;
+    int getMaxPathSum(TreeNode* root){
+        if(!root) return 0;
+        int L = getMaxPathSum(root->left);
+        int R = getMaxPathSum(root->right);
+        maxSum = max(maxSum, L + R + root->val);
+
+        return max(0, max(L, R) + root->val);
+
+    }
+
+    int maxPathSum(TreeNode* root) {
+        maxSum = -1000000;
```

Comment:
`INT_MIN` is always safe. What if the lowest value in the tree node is less than -10^6?